### ISSUE TITLE
Incorrect link

### ISSUE THREAD

**Issue by eitanwa** [WWBN/AVideo#6589]
hey

When an email is received about a live broadcast
A link has been sent
The link does not refer to the live broadcast

For example, this is the link sent to the email
https://alive528.com/live/0/News

The actual broadcast took place on the above link
https://alive528.com/live/0/News/1

eitan live528

**Comment by DanielnetoDotCom** [comment]
I see, 

the problem here is.

you are allowing your users to make more than one live stream with the same key, so each live stream key will have a different/automatic index
when you request a notification I do not know what "index" you will use

but I will try to detect if you are online on the default index, if you are not I will try to see if you have any index online. if there is only one, it will be automatically redirected. 

if there is more that one I will let you choose

**Comment by eitanwa** [comment]
Hey
I will try to clarify
When the mouse cursor is in the area of the video
The page cannot be scrolled up or down
Using the mouse scroll


What I'm trying to show in the picture using the finger icon
The area where the page cannot be scrolled
And that's from the video page and also from the broadcast page


![image](https://user-images.githubusercontent.com/85537931/168636003-8887395f-6514-4a0d-987c-5436116ae7b4.png)


https://alive528.com/live/0/News

https://alive528.com/v/3534?page=1

**Comment by guymass** [comment]
Hi Daniel, please let me know when to update so we can check the mail notification again. Strange we did't notice this till today. I think the notification should take the URL from the live stream once it is on and send that URL.

**Comment by DanielnetoDotCom** [comment]
Hi

I just sent an update that may redirect you in case the URL key is offline and you have another live from the same user

**Comment by JoshWho** [comment]
I am having this same issue now after the last update. all my streams are redirecting to the wrong place.

**Comment by DanielnetoDotCom** [comment]
> I am having this same issue now after the last update. all my streams are redirecting to the wrong place. #6594
> 
> > Something is making all the live streams redirect to the wrong channel/
> > https://www.joshwhotv.com/live/1/Ohiopoliticalnews
> > https://www.joshwhotv.com/live/1/TulsaLiveStream
> > https://www.joshwhotv.com/live/1/DreagusProd/1

As I could see your streams are playing

**Comment by DanielnetoDotCom** [comment]
Yes, the email URL will still wrong, but once you access the URL it may redirect you to the correct one

**Comment by DanielnetoDotCom** [comment]
yes, as I told you before, the email will work (even with the wrong URL)

**Comment by JoshWho** [comment]
> > I am having this same issue now after the last update. all my streams are redirecting to the wrong place. #6594
> > > Something is making all the live streams redirect to the wrong channel/
> > > https://www.joshwhotv.com/live/1/Ohiopoliticalnews
> > > https://www.joshwhotv.com/live/1/TulsaLiveStream
> > > https://www.joshwhotv.com/live/1/DreagusProd/1
> 
> As I could see your streams are playing

Click the Ohio Political one. It redirects. The other ones look like they corrected themselves.

**Comment by DanielnetoDotCom** [comment]
@JoshWho Please update and try again I found something

**Comment by DanielnetoDotCom** [comment]
not really supposed to.

did you clear the cache?

**Comment by JoshWho** [comment]
Yes that is after the cache clear

**Comment by DanielnetoDotCom** [comment]
Josh I did update your site, please try now.

**Comment by JoshWho** [comment]
> Josh I did update your site, please try now.

Working good now. Thank you @DanielnetoDotCom

**Comment by DanielnetoDotCom** [comment]
Just sent another fix, please update

**Comment by guymass** [comment]
Hi Daniel, I have updated today, however in our situation we still get the wrong URL being sent in the email to the subscribers of a channel. When they click the URL it will not open the live stream. How can we overcome this problem?

**Comment by DanielnetoDotCom** [comment]
I am expecting the wrong URL to be sent in the email. but I am also expecting the system to recognize and redirect when the same user is online.

does it not happen? 

how can I test it? it is working on my side.

**Comment by guymass** [comment]
Is the option for multiple streams on? In our case this is due to this option being on to allow a channel to have multiple streams, so our stream URL will end for example with a 1 but in the email I will get a URL ending with a 0. 

It doesn't matter which live stream we do, either from OBS or even playing a playlist. The first email will arrive with a 0.

However, if we play a second stream, that URL will arrive with the correct number, for example 1 in the end of the URL.

This is a screen shot of the mail a user gets:
![photo_2022-05-24 18 18 45](https://user-images.githubusercontent.com/58882903/170071999-9c640914-a23a-4d66-aa4c-e9c3326a9841.jpeg)

The image URL is this:

https://alive528.com/live/0/BeitHamikdash - notice the 0 zero, which is the first stream, and this URL will not open the stream, however any other new stream will get a 1, 2 ,3 etc... depending on how many streams are made after, and those will be sent with the correct URL like this:

https://alive528.com/live/1/BeitHamikdash

So only the zero will not work.

**Comment by guymass** [comment]
Checking now for example my Live stream URL is this:
https://alive528.com/live/0/Streamer528/1

But it seems that it will arrive in the email like this:
https://alive528.com/live/0/Streamer528

without the 1 in the end.

**Comment by guymass** [comment]
You have a user on my site, try to live stream from your channel and see what is happening because its hard to explain really, and I think you should try two stream simultaneously so you can see see what is going on.

**Comment by guymass** [comment]
Also in streaming a playlist this is the URL of the playlist from the home page notification:
https://alive528.com/live/0/Streamer528?playlists_id_live=100926

But in the email I will get this URL:

https://alive528.com/live/0/Streamer528 - which doesn't work because its wrong

**Comment by DanielnetoDotCom** [comment]
I just clicked on the link https://alive528.com/live/0/Streamer528 and I was redirected to https://alive528.com/live/0/Streamer528/1, so I assume it is working

**Comment by guymass** [comment]
But were you able to view the stream? because I couldn't. Yes it redirects at times to the correct URL but the stream was not playing for me. I will do some more testing today, but from what I can say this only happens for the first stream. I am going to start a stream in few min and post here the URL that is sent in the email and please let me know if you can see it. A screen shot would be nice.

**Comment by guymass** [comment]
I have now started a stream and will keep it running for a while for testing.

This is the URL I got in the email:

https://alive528.com/live/0/Streamer528

**Comment by guymass** [comment]
OK after I reset my key I now can see it properly, however I didn't get the mail with the URL :-|
![Screen Shot 5782-09-24 at 18 44 19](https://user-images.githubusercontent.com/58882903/170303412-499be451-0508-4ccf-9abd-2a339380d45b.jpg)
![Screen Shot 5782-09-24 at 18 44 27](https://user-images.githubusercontent.com/58882903/170303450-d50d8cb9-9453-46ef-8205-080474f29e6a.jpg)

**Comment by guymass** [comment]
OK did another test now, the stream is from OBS and I can see it on the front page and access it, this is OK.

I also now got the email and clicking the link it will take me here:
https://alive528.com/live/0/Streamer528

This will not show the live stream. I guess this only happens on the first stream for some reason. 

This is what I get:
![Screen Shot 5782-09-24 at 20 07 54](https://user-images.githubusercontent.com/58882903/170321220-dfdba01e-01f1-4491-89fd-fe54f47b018f.jpg)

So when ever a channel goes live for the first time, may per day IDK, the email URL is sent incomplete and people cannot access it from the email, they will however be able to enter it from the Live notification on the home page.

**Comment by guymass** [comment]
OK now there's a new issue with this as new streams are always use the URL without the leading /1
This causes the to use a URL like this:

https://alive528.com/live/0/BeitHamikdash

![Screen Shot 5782-09-24 at 21 28 58](https://user-images.githubusercontent.com/58882903/170339952-279f9bed-2db0-46c1-8afb-48f1eff492db.jpg)

And this page doesn't play the live stream.

Trying to add the leading /1 will always reset it back to the URL without a /1

This is an issue that effects all channels and all first live streams on our site. So i guess this /1 is effecting the way first initiated live streams URL is handled.

**Comment by DanielnetoDotCom** [comment]
From your pages, I am getting ERR_TOO_MANY_REDIRECTS error

are your streams really live?

for example, check the demo site.

if you try to access https://demo.avideo.com/live/0/MovieclipsTrailers/1 you will be redirected to https://demo.avideo.com/live/0/MovieclipsTrailers

**Comment by guymass** [comment]
Well it was live for a while but eventually I stopped it because I had to go out. Yes this error was because that first live was stuck even after it was ended.

**Comment by DanielnetoDotCom** [comment]
are you saying you are not being redirected?

Links 

1. https://demo.avideo.com/live/0/MovieclipsTrailers/1
2. https://demo.avideo.com/live/0/MovieclipsTrailers

can you please click on the link (2) link and then click on the link (1) again? 

for some reason my link (2) some times is marked as finished

but in your case, we have the worst error, the ERR_TOO_MANY_REDIRECTS

I am not sure where this error comes from

**Comment by guymass** [comment]
Your is redirecting fine, mine also redirects but it doesn't show the stream. I will do some more checks in couple of hours when I am back home and see if this persists. Currently I checked with 3 different channels and they all had the same issue.

**Comment by guymass** [comment]
OK I did an update just now and done a new stream test and it seems to be working now, don't know what you changed but much thanks. I will keep testing with other channels and if they all can be viewer with the URL sent in the email than I think this issue is now fixed.

**Comment by guymass** [comment]
OK we have tested another channel and this time we ran two streams from it. Just a question, does the second stream is created as a playlist? because the second stream has a URL like this:

https://alive528.com/live/0/News?playlists_id_live=1653591955

**Comment by guymass** [comment]
Another test we just did was to start a stream directly from the site and then start another stream from OBS. The second stream from OBS stopped the first stream.

**Comment by guymass** [comment]
Another issue is when streaming a playlist. Starting a Live stream of a playlist cannot be stopped and when its not even playing it will still show that it is live:
![photo_2022-05-26 22 40 34](https://user-images.githubusercontent.com/58882903/170564846-1eebd7ab-2dca-49fa-85db-3485785dc0bf.jpeg)

The playlist doesn't have a stop button, so to stop it we start a second live playlist and stop it and that will also stop the first one.

So I think there is still some issues with live streaming  from different methods like WebCam, OBS or playlist, each will behave differently and sometime will get a playlist stuck.

**Comment by guymass** [comment]
This playlist for example has stoped since last night, but still this morning is showing on the front page, there is no way for me to clear it or stop it, so now I have to start another stream and immediately stop it and this will also stop and remove this stuck playlist. 

Notice this is the same "NASA" playlist form last night:

![Screen Shot 5782-09-26 at 8 59 18](https://user-images.githubusercontent.com/58882903/170639663-5c0eae05-e539-4c19-9e89-16c317997cf4.jpg)

**Comment by DanielnetoDotCom** [comment]
> https://alive528.com/live/0/News?playlists_id_live=1653591955

that means it comes from a playlist. AGAIN, this feature still experimental

**Comment by DanielnetoDotCom** [comment]
# The playlist live feature still experimental

**Comment by guymass** [comment]
OK thanks I understand. Regardless and putting aside those live playlists, is the feature of multiple live streams experimental as well?
